test/sql/index/art/scan/test_in_filter_index_scan.test

```diff
@@ -0,0 +1,79 @@
+# name: test/sql/index/art/scan/test_in_filter_index_scan.test
+# description: Test using an index scan for IN filter pushdown.
+# group: [scan]
+
+statement ok
+CREATE TABLE tbl AS SELECT range AS i FROM range(500000);
+
+statement ok
+ALTER TABLE tbl ADD PRIMARY KEY(i);
+
+query II
+EXPLAIN ANALYZE DELETE FROM tbl WHERE i IN (3, 50, 299, 123);
+----
+analyzed_plan	<REGEX>:.*Type: Index Scan.*
+
+query I
+SELECT COUNT(*) FROM tbl;
+----
+499996
+
+# Test with duplicate values.
+
+query II
+EXPLAIN ANALYZE SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101);
+----
+analyzed_plan	<REGEX>:.*Type: Index Scan.*
+
+query I
+SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101) ORDER BY i;
+----
+2
+42
+100
+101
+
+# Test with different CONJUNCTION_AND removing values from the IN.
+
+query II
+EXPLAIN ANALYZE SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101) AND i != 42 AND i <= 100;
+----
+analyzed_plan	<REGEX>:.*Type: Index Scan.*
+
+query I
+SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101) AND i != 42 AND i <= 100 ORDER BY i;
+----
+2
+100
+
+query II
+EXPLAIN ANALYZE SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101) AND i = 42 AND i <= 100;
+----
+analyzed_plan	<REGEX>:.*Type: Index Scan.*
+
+query I
+SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101) AND i = 42 AND i <= 100 ORDER BY i;
```

Comment:
What about a test like 
```
SELECT i FROM tbl WHERE i IN (2, 42, 100, 42, 101) OR i = 50 OR i = 100 ORDER BY I;
```
But I think this might be better to handle in `filter_combiner.cpp`

Reply:
That's a cool idea - I've linked it to the follow-up issue for this PR. :)